test_apps/smokeTests/testGrid.cpp

```diff
@@ -0,0 +1,274 @@
+        exit(1);
+    }
+   
+    if (opt.extents.size() != 6 ) {
+        cerr << "The -extents flag must contain 6 elements if used" << endl;
+        op.PrintOptionHelp(stderr, 80, false);
+        exit(1);
+    }
+
+    if (opt.dims.size() != 3) {
+        cerr << "The -dims flag must contain 3 elements if used" << endl;
+        op.PrintOptionHelp(stderr);
+        exit(0);
+    }
+
+    if (opt.help) {
+        op.PrintOptionHelp(stderr);
+        exit(0);
+    }
+}
+
+
+int main( int argc, char** argv ) {
+    double t0 = Wasp::GetTime();
+
+    OptionParser op;
+    InitializeOptions( argc, argv, op );
+
+    std::cout << std::fixed << std::showpoint;
+    std::cout << std::setprecision(4);
+
+    std::vector< double > minu = { 
+        opt.extents[X], 
+        opt.extents[Y], 
+        opt.extents[Z] 
+    };
+    std::vector< double > maxu = { 
+        opt.extents[X+3], 
+        opt.extents[Y+3], 
+        opt.extents[Z+3] 
+    };
+
+    std::vector< size_t > bs2d = { opt.bs[X], opt.bs[Y] }; 
+    std::vector< double > minu2d = { minu[X], minu[Y] };
+    std::vector< double > maxu2d = { maxu[X], maxu[Y] };
+
+    std::vector< size_t > dims2d = { opt.dims[X], opt.dims[Y] };
+
+    int rc            = 0;
+    int regularRC     = 0;
+    int stretchedRC   = 0;
+    int layeredRC     = 0;
+    int curvilinearRC = 0;
+
+    // Test Regular Grid
+    if ( std::find( opt.grids.begin(), opt.grids.end(), "Regular" ) != opt.grids.end() ) {
+        std::vector< float* > rgBlks  = AllocateBlocks( opt.bs, opt.dims );
+        RegularGrid* regularGrid = new RegularGrid( opt.dims, opt.bs, rgBlks, minu, maxu );
+        regularRC = RunTests( regularGrid, opt.arrangements, opt.minValue, opt.maxValue );
+        delete regularGrid;
+    }
+
+    // Test Stretched Grid
+    if ( std::find( opt.grids.begin(), opt.grids.end(), "Stretched" ) != opt.grids.end() ) {
+        StretchedGrid* stretchedGrid = MakeStretchedGrid( opt.dims, opt.bs, minu, maxu );
+        stretchedRC = RunTests( stretchedGrid, opt.arrangements, opt.minValue, opt.maxValue );
+        delete stretchedGrid;
+    }
+
+    // Test Layered Grid
+    if ( std::find( opt.grids.begin(), opt.grids.end(), "Layered" ) != opt.grids.end() ) {
+        LayeredGrid* layeredGrid = MakeLayeredGrid( opt.dims, opt.bs, minu, maxu );
+        layeredRC = RunTests( layeredGrid, opt.arrangements, opt.minValue, opt.maxValue );
+        delete layeredGrid;
+    }
+   
+    // Test Curvilinear Grid
+    if ( std::find( opt.grids.begin(), opt.grids.end(), "Curvilinear" ) != opt.grids.end() ) {
+        CurvilinearGrid* curvilinearGrid;
+        curvilinearGrid = MakeCurvilinearTerrainGrid( opt.bs, minu, maxu, opt.dims);
+        curvilinearRC = RunTests( curvilinearGrid, opt.arrangements, opt.minValue, opt.maxValue );
+        delete curvilinearGrid;
+    }
+
+    if ( regularRC != 0 ) {
+        cout << "Errors occurred while testing Grid::RegularGrid." << endl;
+        rc = -1;
```

Comment:
Follow exit code conventions for programs; return 0 on success and 1 on failure

Reply:
Thanks - I confused return codes with exit codes here.